src/ims.js

```diff
@@ -419,22 +425,80 @@ class Ims {
   }
 
   /**
-   * Verifies a given token.
+   * Validates the given token against an allow list.
+   *
+   * Optional: If a cache is provided, the token will be validated against the cache first.
+   *
+   * Note: The cache uses the returned status key to determine if the result should be cached. This is not returned
+   *       to the user.
+   *
+   * @param {*} token the token to validate
+   * @param {*} allowList the allow list to validate against
+   * @returns {Promise} Promise that resolves with the ims validation result
+   */
+  async validateTokenAllowList (token, allowList) {
```

Comment:
nit: we should mention type of allowList as array in JS doc comments

Reply:
Good catch, fix incoming